### PR TITLE
feat: Fetch Proposal Data and Display in Governance List

### PR DESCRIPTION
**Pull request by damiano-melcarne-lab49**
**Description**:

- Fetching all proposals from events emitted from the smart contract through a mirror node API endpoint.
Screen Shot 2022-11-17 at 10 54 51 PM
- Created horizontal stacked bar chart component for vote count display.
Screen Shot 2022-11-17 at 10 50 47 PM
- Converted proposal block numbers to human readable time durations.
Screen Shot 2022-11-17 at 10 50 54 PM

**Fixes**

- Updated `HederaService` read-only contract calls to use `ContractCallQuery` instead of `ContractExecuteTransaction`. This reduces the hbar costs of executing these queries.

### CODE REVIEW

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 1]
```diff
@@ -40,17 +42,22 @@ function createHederaService() {
     return _precision;
   };
 
+  const queryContract = async (
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:45]: `ContractCallQuery` calls were consolidated into a single function to simplify the HederaService code and apply consistent gas payment amounts.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 2]
```diff
@@ -252,50 +300,31 @@ function createHederaService() {
 
   // TODO: will need to pass in contractId in future when there are more pools
   const pairCurrentPosition = async (_contractId: ContractId = contractId) => {
-    const getPairQty = new ContractExecuteTransaction()
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:255]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 3]
```diff
@@ -252,50 +300,31 @@ function createHederaService() {
...
-    console.log(`${tokenAQty} units of token A and ${tokenBQty} units of token B are present in the pool. \n`);
+    const result = await queryContract(_contractId, PairContractFunctions.GetPoolBalances);
+    const tokenAQty = result?.getInt256(0);
+    const tokenBQty = result?.getInt256(1);
     // TODO: dont hardcodethis, will have to be dynamic
     return { [TOKEN_A_SYMBOL]: tokenAQty, [TOKEN_B_SYMBOL]: tokenBQty };
   };
 
-  const getContributorTokenShare = async () => {
-    const getContributorTokenShare = new ContractExecuteTransaction()
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:270]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 4]
```diff
@@ -252,50 +300,31 @@ function createHederaService() {
...
+  }> => {
+    const queryParams = new ContractFunctionParameters().addAddress(treasuryId.toSolidityAddress());
+    const result = await queryContract(contractId, PairContractFunctions.GetLiquidityProviderTokenAmounts, queryParams);
+    const tokenAQty = result?.getInt64(0);
+    const tokenBQty = result?.getInt64(1);
+    return { tokenAQty, tokenBQty };
   };
 
-  const getSpotPrice = async () => {
-    const getSpotPrice = new ContractExecuteTransaction()
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:287]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 5]
```diff
@@ -305,75 +334,42 @@ function createHederaService() {
   };
 
   const fetchFee = async (): Promise<BigNumber | undefined> => {
-    const getFee = new ContractExecuteTransaction()
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:308]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 6]
```diff
@@ -305,75 +334,42 @@ function createHederaService() {
...
-    const response = await getPoolFeeTransaction.getRecord(client);
-    const fee = response.contractFunctionResult?.getInt256(0);
+    const result = await queryContract(contractId, PairContractFunctions.GetFee);
+    const fee = result?.getInt256(0);
     const feePrecision = await fetchFeePrecision();
     return fee?.div(feePrecision?.toNumber() ?? 1);
   };
 
   const fetchFeePrecision = async (): Promise<BigNumber | undefined> => {
-    const getFeePrecision = new ContractExecuteTransaction()
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:321]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 7]
```diff
@@ -305,75 +334,42 @@ function createHederaService() {
...
+    const result = await queryContract(contractId, PairContractFunctions.GetFeePrecision);
+    const feePrecision = result?.getInt256(0);
     return feePrecision;
   };
 
   const getTokenBalances = async (): Promise<{
     amountOfTokenA: BigNumber | undefined;
     amountOfTokenB: BigNumber | undefined;
   }> => {
-    const getTokenBalance = new ContractExecuteTransaction()
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:336]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 8]
```diff
@@ -305,75 +334,42 @@ function createHederaService() {
...
-    const response = await getTokenBalanceTx.getRecord(client);
-    const amountOfTokenA = response.contractFunctionResult?.getInt256(0);
-    const amountOfTokenB = response.contractFunctionResult?.getInt256(1);
+    const result = await queryContract(contractId, PairContractFunctions.GetPoolBalances);
+    const amountOfTokenA = result?.getInt256(0);
+    const amountOfTokenB = result?.getInt256(1);
     return { amountOfTokenA, amountOfTokenB };
   };
 
-  const getContractAddress = async () => {
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:348]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/services/HederaService/HederaService.ts [hunk 9]
```diff
@@ -305,75 +334,42 @@ function createHederaService() {
...
-      response.contractFunctionResult?.getAddress(0) ?? ""
-    ).toString();
-    console.log(contractAddress);
+  const getContractAddress = async (): Promise<string> => {
+    const result = await queryContract(contractId, PairContractFunctions.GetContractAddress);
+    const contractAddress = AccountId.fromSolidityAddress(result?.getAddress(0) ?? "").toString();
+    return contractAddress;
   };
 
-  const getTokenPairAddress = async () => {
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/HederaService.ts:362]: Refactored to use the simplified `queryContract` function.

File: src/dex-ui/pages/Governance/ProposalCard.tsx [hunk 10]
```diff
@@ -53,20 +72,41 @@ export const ProposalCard = (props: ProposalCardProps) => {
...
-            <HStack flex="1">
-              <Circle size="1em" bg="#D1CED1" />
-              <Text>{proposal.voteCount.abstain}</Text>
-            </HStack>
-          </Flex>
+          {proposal.state === ProposalState.Active ? (
+            <VStack spacing={2} align="stretch">
+              <Metrics
+                flex="1"
+                data={[
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/Governance/ProposalCard.tsx:79]: Using the DEX v2 design colors.

**Review by roshan-bisht-lab49 — APPROVED** [review]

**Pull request merged**